src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCompiler.cs

```diff
@@ -4904,19 +4919,19 @@ private void GenerateOneCode()
                             Ldloc(_runtextbegLocal!);
                         }
                         Sub();
-                        if (c != int.MaxValue)
-                        {
-                            Label l4 = DefineLabel();
-                            Dup();
-                            Ldc(c);
-                            Blt(l4);
-                            Pop();
-                            Ldc(c);
-                            MarkLabel(l4);
-                        }
                         using (RentedLocalBuilder cLocal = RentInt32Local())
                         {
                             Stloc(cLocal);
+                            if (c != int.MaxValue)
```

Comment:
Aside - the commented blocks of equivalent code seem to be pretty out of sync in some cases. Eg., `int.MaxValue` doesn't appear in the block above. I wonder whether it is worth (separately) updating those as far as possible. It is generally not possible to infer what is going on by comparing the interpreter.

Reply:
Yeah, I've added and cleaned up existing comments as I go, but there's still a non-trivial amount of code in the library that could stand to have better / revised comments.